packages/components/src/components/Icon/Icon.tsx

```diff
@@ -42,8 +44,16 @@ const Icon = React.forwardRef<HTMLDivElement, IconProps>(
       throw new Error(`${icon}: Missing a title for non-decorative icon.`);
     }
 
-    // eslint-disable-next-line import/namespace
-    const RenderedIcon = HeroOutlineIcons[icon];
+    let RenderedIcon = undefined;
+
+    if (isHeroIcon(icon)) {
+      // eslint-disable-next-line import/namespace
+      RenderedIcon = HeroOutlineIcons[icon];
+    }
+
+    if (isLucideIcon(icon)) {
+      RenderedIcon = LucideIcons[icon];
+    }
```

Comment:
No need to check the second `if` if the first `if` is satisfied.
```suggestion
    if (isHeroIcon(icon)) {
      // eslint-disable-next-line import/namespace
      RenderedIcon = HeroOutlineIcons[icon];
    } else if (isLucideIcon(icon)) {
      RenderedIcon = LucideIcons[icon];
    }
```